nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/serving/SuggestionOp.java

```diff
@@ -77,27 +80,40 @@
     @Context
     protected MarshallerRegistry registry;
 
-    @Param(name = "document", description = "A document", required = false)
-    protected DocumentModel documentModel;
+    @Param(name = "updatedDocument", description = "Document with changes done on client side before saving", required = false)
+    protected DocumentModel updatedDocument;
 
     @Param(name = "references", description = "Should the entity references be resolved?", required = false)
     protected boolean references = false;
 
+    @OperationMethod
+    public Blob run(DocumentRef docRef) {
+        DocumentModel docModel = coreSession.getDocument(docRef);
+        return run(docModel);
+    }
+
     @OperationMethod
     public Blob run(DocumentModel doc) {
+        if (updatedDocument != null) {
+            Arrays.asList(doc.getSchemas())
+                  .forEach(schema -> updatedDocument.getProperties(
+                          schema).forEach((key, value) -> doc.setPropertyValue(key, (Serializable) value)));
+        }
 
-        List<EnrichmentMetadata> suggestions;
-        if (doc == null || (suggestions = modelServingService.predict(doc)) == null || suggestions.isEmpty()) {
```

Comment:
Not sure why you changed this?  The current style was suggested in a code review

Reply:
cleaner, I'd assume

Reply:
Two reasons:
- Initially for my mind map and for debugging;
- Then I found cleaner, more readable;

But I can change it to the original one.

Reply:
Leave it, how you did it.